src/generator.cs

```diff
@@ -4850,6 +4850,21 @@ void GenerateAsyncMethod (MemberInformation original_minfo, AsyncMethodKind asyn
 		var mi = original_minfo.method;
 		var minfo = new AsyncMethodInfo (this, original_minfo.type, mi, original_minfo.category_extension_type, original_minfo.is_extension_method);
 		var is_void = mi.ReturnType == TypeManager.System_Void;
+
+		// Print a error if any of the method parameters or handler parameters is ref/out, it should not be asyncified.
+		if (minfo.async_initial_params != null) {
+			foreach (var param in minfo.async_initial_params) {
+				if (param.ParameterType.IsByRef) {
+					throw new BindingException (1062, true, $"The member '{original_minfo.type.Name}.{mi.Name}' contains ref/out parameters and must not be decorated with [Async].");
```

Comment:
Do we not want to put that into a variable so we can share it with https://github.com/xamarin/xamarin-macios/pull/3059/files#diff-8daf8335e79369d64ff3c1ac53402076R4864?

I'm thinking that if we ever update the exception message we could forget one.